### DISCUSSION TITLE
img2img Hires fix

### DISCUSSION THREAD

**Discussion by barepixels** [lllyasviel/stable-diffusion-webui-forge#1669] [General]
Just want to share what I found today (tested and work in Forge)

https://github.com/Amadeus-AI/img2img-hires-fix
https://www.reddit.com/r/StableDiffusion/comments/1f706zg/i_made_an_img2img_hires_fix_webui_extension/

Hires fix upscale right after the img2img process, using its latent.

Inspired by wcde/custom-hires-fix-for-automatic1111, which has been archived and isn't functioning since 2023.

Simple 150 lines extension with minimized features and dependency.

It also works in text2img, you can check the consistency between the Original "Hires. fix" and this extension.

I made this because I need this feature🤣

(Originally, if you upscale using/before img2img, the anatomy will be broken at higher resolution.)

**Comment by angelika3377** [comment]
thank you, I needed it too!

**Comment by dilectiogames** [comment]
well look at that. I wish I had the same luck at money and women.

this fixed my issue that I've just posted in the bug tab. 🤣🤣🤣🤣

https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/2565

I've had only one small issue that the custom steps value are capped at 25 but I've fixed that myself in the script at line 44 change _"maximum=25"_ to 150 to be the same as the vanilla Hires.fix (even if I never used a number that high)

**Comment by Moussack** [comment]
There's no latent (nearest exact) for the upscaler, am i missing smt??

**Comment by SvenErik1968** [comment]
@barepixels, if you haven't yet done so, please add your extension to the [Forge Extension List](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1754) so more people will see this (it is linked to from the front page here).

**Comment by strongshuai** [comment]
![68747470733a2f2f692e696d6775722e636f6d2f6b6f4f324535712e6a706567](https://github.com/user-attachments/assets/1a705530-e66b-4c79-b88f-a70a1a484fbf)
![68747470733a2f2f692e696d6775722e636f6d2f627259355159772e6a706567](https://github.com/user-attachments/assets/e3d77b71-3255-43e4-af97-69a4f0498977)
Is this a high-definition zoom comparison image? Why is there not much difference except for the hair color?

**Reply by HMRMike** [reply]
These two images are not a comparison but a source and result.
Compare to shit result without the fix
![68747470733a2f2f692e696d6775722e636f6d2f72484a474c75712e6a706567](https://github.com/user-attachments/assets/c6a605cb-3407-490a-abcc-26b2cc9acfdf)
